### PR TITLE
#7 create an endpoint to get actual materials

### LINKED ISSUE #7: Create an endpoint to get actual materials

### PR DESCRIPTION
**Pull request by Ola-M**
Closes #7

### CODE REVIEW

File: TribalWars/src/WebApi/Hubs/VillageMaterialsHub.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            await AddToVillageMaterialsGroup(GroupType.VillageMaterials.ToString());
+        }
+
+        public async Task UpdateVillageMaterials(IEnumerable<VillageMaterial> actualVillageMaterials)
+        {
+           await _villageMaterialServices.UpdateVillageMaterials(actualVillageMaterials);
+        }
+
+        [Authorize]
+        public async Task<IEnumerable<VillageMaterialVM>> BartekZDrungiemZaPociungiem()
```
Comment (HerringTheCoder) [inline TribalWars/src/WebApi/Hubs/VillageMaterialsHub.cs]: Very expressive method name. I like it.

File: TribalWars/src/WebApi/Workers/VillageMaterialsTimerWorker.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            _hubContext = hubContext;
+            _localizer = localizer;
+        }
+
+        protected async override Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                await _hubContext.Clients.Group(GroupType.VillageMaterials.ToString())
+                    .RefreshVillageMaterialst(_localizer["RefreshVillageMaterials"]);
```
Comment (HerringTheCoder) [inline TribalWars/src/WebApi/Workers/VillageMaterialsTimerWorker.cs]: RefreshVillageMaterials*

File: TribalWars/src/WebApi/Hubs/VillageMaterialsHub.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Localization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebApi.Hubs
+{
+    public class VillageMaterialsHub : Hub<IVillageMaterialsClient>
```
Comment (HerringTheCoder) [inline TribalWars/src/WebApi/Hubs/VillageMaterialsHub.cs:17]: I think most of these actions should be authorized. You can put [Authorize] annotation on top of the Hub class, if all methods are supposed to be protected.

**Review by HerringTheCoder — CHANGES_REQUESTED** [review]

File: TribalWars/src/WebApi/Hubs/VillageMaterialsHub.cs (hunk 1 above)
Comment (Skrabka98) [reply TribalWars/src/WebApi/Hubs/VillageMaterialsHub.cs]: All I can say about it is WTF?!?

**Pull request merged**